src/main/java/xbot/common/injection/DevicePolice.java

```diff
@@ -91,6 +91,37 @@ public String registerDevice(DeviceType type, int id, Object device) {
         return entry;
     }
 
+    /**
+     * Register a CAN device
+     * @param type Device type
+     * @param busId Device CAN bus id
+     * @param id Device id
+     * @param device Device object
+     */
+    public String registerDevice(DeviceType type, CANBusId busId, int id, Object device) {
+        // First, check to see if the overall device has already been registered once. We only
+        // want there to be one "main" entry for a given device, like an Encoder, which may use two channels.
+        // That way, when we ask the DevicePolice how many devices have been registered, it wouldn't return the
+        // same Encoder twice.
+        if (!registeredDevices.contains(device)) {
+            registeredDevices.add(device);
+        }
+
+        if (type != DeviceType.CAN) {
```

Comment:
⭐ should this be at the top of the method instead of after the other stuff happens?